src/couch_quickjs/test/couch_quickjs_tests.erl

```diff
@@ -60,9 +63,55 @@ t_bad_memory_limit(_) ->
     ?assertEqual(ExpectCmd, string:find(Cmd, ExpectCmd)),
     ?assertEqual(1, os_cmd(Cmd ++ " -V")).
 
+t_couch_jsengine_config_triggers_proc_server_reload(_) ->
+    case couch_server:with_spidermonkey() of
+        false ->
+            % Spidermonkey is not in the build at all, skip the test
+            ok;
+        true ->
+            OldVal = get_proc_manager_default_js(),
+            % In the test, set the engine to whatever is not the default
+            Toggle =
+                case couch_server:get_js_engine() of
+                    <<"quickjs">> -> "spidermonkey";
+                    <<"spidermonkey">> -> "quickjs"
+                end,
+
+            config:set("couchdb", "js_engine", Toggle, false),
+            % couch_server:get_js_engine/0 should be visible immediately
+            ?assertEqual(list_to_binary(Toggle), couch_server:get_js_engine()),
+
+            wait_until_proc_manager_updates(OldVal),
+            ?assertNotEqual(OldVal, get_proc_manager_default_js()),
+            NewVal = get_proc_manager_default_js(),
+
+            % Toggle back to the original default (test config:delete/3)
+            config:delete("couchdb", "js_engine", false),
+            wait_until_proc_manager_updates(NewVal),
+            ?assertNotEqual(NewVal, get_proc_manager_default_js()),
+            % We should be back to the original default
+            ?assertEqual(OldVal, get_proc_manager_default_js())
+    end.
+
 os_cmd(Cmd) ->
     Opts = [stream, {line, 4096}, binary, exit_status, hide],
     Port = open_port({spawn, Cmd}, Opts),
     receive
         {Port, {exit_status, Status}} -> Status
     end.
+
+wait_until_proc_manager_updates(OldVal) ->
+    WaitFun = fun() ->
+        case get_proc_manager_default_js() of
+            OldVal -> wait;
+            _ -> ok
+        end
+    end,
+    case test_util:wait(WaitFun, 5000) of
```

Comment:
The default timeout for an individual test is [5 seconds](https://www.erlang.org/doc/apps/eunit/chapter.html#control).
Running the test takes time, and the wait function can take up to 5 seconds.
Adding them together would exceed 5 seconds and cause `timeout`.
Shall we use a smaller number, like `4500`?

Reply:
Works as expected.
```erl
(node1@127.0.0.1)1> couch_server:get_js_engine().
<<"quickjs">>
(node1@127.0.0.1)2> config:set("couchdb", "js_engine", "spidermonkey").
ok
(node1@127.0.0.1)3> couch_server:get_js_engine().
<<"spidermonkey">>
(node1@127.0.0.1)4> config:set("couchdb", "js_engine", "quickjs").
ok
(node1@127.0.0.1)5> couch_server:get_js_engine().
<<"quickjs">>
```

Just curious what happens if I make a typo... Is that possible to avoid this error?
```bash
(node1@127.0.0.1)6> config:set("couchdb", "js_engine", "spider_monkey").
ok
*** ERROR: Shell process terminated! (^G to start new job) ***

$ curl $DB
curl: (7) Failed to connect to 127.0.0.1 port 15984 after 0 ms: Couldn't connect to server
```

Reply:
Currently it's somewhat intentional that it would crash on function clause on an invalid engine setting. But perhaps it's better to emit an error in the logs and then use the default? But I think that's for a separate PR.

Reply:
Made a PR to handle it better : just stick with the default value